azurelinuxagent/common/osutil/ubuntu.py

```diff
@@ -39,10 +39,18 @@ def start_network(self):
         return shellutil.run("service networking start", chk_err=False)
 
     def stop_agent_service(self):
-        return shellutil.run("service {0} stop".format(self.service_name), chk_err=False)
+        try:
+            shellutil.run_command(["service", self.service_name, "stop"])
+        except shellutil.CommandError as cmd_err:
+            return cmd_err.returncode
+        return 0
```

Comment:
The callers don't do anything with the return value, same for `start_agent_service` below. I see it's existing behaviour, up to you if you want to change it now or in a following PR.

Reply:
Let's not introduce any changes of behavior in this PR (unless the previous behavior is incorrect)